internal/charmstore/store.go

```diff
@@ -742,53 +742,65 @@ func (s *Store) UpdateBaseEntity(url *router.ResolvedURL, update bson.D) error {
 	return nil
 }
 
+var ErrPublishResourceMismatch = errgo.Newf("charm published with incorrect resources")
+
 // Publish assigns channels to the entity corresponding to the given URL.
 // An error is returned if no channels are provided. For the time being,
 // the only supported channels are "development" and "stable".
```

Comment:
// If a resource is provided that is not specified in the charm metadata an error with the cause ErrPublichResourceMismatch will be returned.